src/test/java/com/coherent/testng/JsonParserTNG.java

```diff
@@ -0,0 +1,91 @@
+package com.coherent.testng;
+
+import com.coherent.junit.MyParser;
+import org.junit.jupiter.api.Assertions;
+import org.testng.annotations.*;
+import parser.JsonParser;
+import parser.NoSuchFileException;
+import shop.Cart;
+import shop.RealItem;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+public class JsonParserTNG {
+
+    Cart allaCart = new Cart("alla-cart-testNG");
+    JsonParser jsonParser = new JsonParser();
+
+    @BeforeMethod
+    public void createCart() {
+        RealItem newItem = new RealItem();
+        newItem.setName("Item A");
+        newItem.setWeight(100);
+        newItem.setPrice(25);
+        allaCart.addRealItem(newItem);
+        jsonParser.writeToFile(allaCart);
+    }
+
+
+    //TODO Question!
+
+    @Ignore
+    @Test(groups = {"JsonParserTest.regular"})
+    public void checkReadsFromFileCorrectly() {
+        int totalPrice = 30;
+        //This test fails. Looks like @AfterMethod is not working properly? TBD
+        Cart allaCart = MyParser.readFromFile(new File("src/main/resources/alla-cart-testNG.json"));
+        allaCart.getTotalPrice();
+        Assertions.assertEquals(allaCart.getTotalPrice(), totalPrice);
+    }
+
+
+    @Test(groups = {"JsonParserTest.regular"})
+    public void checkFileDeleted() {
+        File fileToDelete = new File("src/main/resources/alla-cart-testNG.json");
+        fileToDelete.delete();
+
+        Assertions.assertEquals(fileToDelete.delete(), false);
+    }
```

Comment:
It is not unit test for parser classes. This test check  method from File class.

Reply:
Method has been deleted